sdk/storage/azure-storage-file/apireview/azure/storage/file/directory_client.pyi

```diff
@@ -0,0 +1,40 @@
+from ._shared.request_handlers import add_metadata_headers
+from ._shared.response_handlers import process_storage_error, return_response_headers
+from .file_client import FileClient
+from .models import ContentSettings, DirectoryProperties, DirectoryPropertiesPaged, HandlesPaged, NTFSAttributes, ShareProperties
+from azure.core.paging import ItemPaged
+from typing import Any, Dict, Optional, Union
+
+class DirectoryClient(StorageAccountHostsMixin):
+    snapshot: Any = ...
+    share_name: Any = ...
+    directory_path: Any = ...
+    def __init__(self, directory_url: str, share: Optional[Union[str, ShareProperties]]=..., directory_path: Optional[str]=..., snapshot: Optional[Union[str, Dict[str, Any]]]=..., credential: Optional[Any]=..., **kwargs: Optional[Any]) -> None: ...
+    @classmethod
+    def from_connection_string(cls: Any, conn_str: str, share: Optional[Union[str, ShareProperties]]=..., directory_path: Optional[str]=..., credential: Optional[Any]=..., **kwargs: Any) -> DirectoryClient: ...
+    def get_file_client(self, file_name: str, **kwargs: Any) -> FileClient: ...
+    def get_subdirectory_client(self, directory_name: str, **kwargs: Any) -> DirectoryClient: ...
+    def create_directory(self, metadata: Optional[Dict[str, str]]=..., timeout: Optional[int]=..., **kwargs: Optional[Any]) -> Dict[str, Any]: ...
+    def delete_directory(self, timeout: Optional[int]=..., **kwargs: Any) -> None: ...
+    def list_directories_and_files(self, name_starts_with: Optional[str]=..., timeout: Optional[int]=..., **kwargs: Any) -> ItemPaged: ...
+    def list_handles(self, recursive: bool=..., timeout: Optional[int]=..., **kwargs: Any) -> ItemPaged: ...
+    def close_handles(self, handle: Union[str, HandleItem]=..., recursive: bool=..., timeout: Optional[int]=..., **kwargs: Any) -> Any: ...
+    def get_directory_properties(self, timeout: Optional[int]=..., **kwargs: Any) -> DirectoryProperties: ...
+    def set_directory_metadata(self, metadata: Dict[str, Any], timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def set_http_headers(self, file_attributes: Union[str, NTFSAttributes]=..., file_creation_time: Union[str, datetime]=..., file_last_write_time: Union[str, datetime]=..., file_permission: Optional[str]=..., file_permission_key: Optional[str]=..., timeout: Optional[int]=..., **kwargs: Any) -> Dict[str, Any]: ...
+    def create_subdirectory(self, directory_name: str, metadata: Optional[Dict[str, Any]]=..., timeout: Optional[int]=..., **kwargs: Any) -> DirectoryClient: ...
+    def delete_subdirectory(self, directory_name: str, timeout: Optional[int]=..., **kwargs: Any) -> None: ...
+    def upload_file(self, file_name: str, data: Any, length: Optional[int]=..., metadata: Optional[Dict[str, str]]=..., content_settings: Optional[ContentSettings]=..., validate_content: bool=..., max_connections: Optional[int]=..., timeout: Optional[int]=..., encoding: str=..., **kwargs: Any) -> FileClient: ...
```

Comment:
Move lots of options parameters into kwargs

Reply:
#7807